documentation/Database-Schema-Application.md

```diff
@@ -0,0 +1,56 @@
+# Database Schema Application
+
+![Database Schema](images/database-schema-application.png)
+
+## Tables description: Application
+
+### Application instance
+The application has all the visualisation and logic defined in the <b>application template</b> [(described in Database Schema Template)](Database-Schema-Template.md). When the Applicant clicks to apply for a application type (aka template) a new instance of one application is created for them. The Applicant user can make changes until submitting the application to be reviewed.
+
+* <b>application</b>
+Representation of the application instance. All nested elements are accessible via joined tables and can be created or queried in the same call using the GraphQL engine.
+The `template_id` links to the template used to display all sections and elements (questions) to the user.
+The `user_id` links to the Applicant user that created this application.
+The `serial` is a number used to display the application number to users. 
+The `name` will be deduced by the template name + user + company names.
+The `is_active` is set to `'true'` for all applications that are still in the review process. 
+The `is_active` is set to `'false'` for applications that have finished the review process, have expire or were withdrawn by the Applicant. 
+The `trigger` is updated everytime the application has changes done by users or a scheduler. The trigger would be one of the options: `'onApplicationCreate'`, `'onApplicationSubmit'`, `'onApplicationSave'`, `'onApplicationWithdrawn'`, `'onReviewStart'`, `'onReviewEditComment'`, `'onReviewSave'`, `'onReviewAssign'`, `'onApprovalSubmit'`, `'onScheduleTime'`.
+The `outcome` is either `'Pending'`, `'Approved'`, `'Rejected'`. The application is 'Pending' during the review process, each stage will store also the outcome, so it will only be updated to here once the review process is finished resulting in either 'Approved' or 'Rejected'.
+<b>To be considered:</b>
+   * The `unique_identifier` is a new field. What is it used for? Should is replace the `serial`?
+   * An application with Draft should be flagged as `is_active`?
+
+* <b>application_section</b>
+Sections of the application that contain some elements. This is exactly the same as the `template_section`.
+The `application_id` links to the application.
+The `template_section_id` links to the template.
+<b>To be considered: Why so we need an `application_section` and not `application_element`. We should use the same logic for both.</b>
```

Comment:
Yes, good point. `application_response` (which is the equivalent of `template_element`) should be related to `application_section` rather than `application` to keep consistency between Template and Application.

You want to go ahead and change it, or will you make a new issue?

Reply:
By what was discussed in today's meeting we will keep as it is.